src/Core/Framework/DataAbstractionLayer/Search/RequestCriteriaBuilder.php

```diff
@@ -53,7 +53,8 @@ public function handleRequest(Request $request, Criteria $criteria, EntityDefini
         if ($request->getMethod() === Request::METHOD_GET) {
             $criteria = $this->fromArray($request->query->all(), $criteria, $definition, $context, $request->attributes->getInt('version'));
         } else {
-            $criteria = $this->fromArray($request->request->all(), $criteria, $definition, $context, $request->attributes->getInt('version'));
+            $params = array_merge($request->query->all(), $request->request->all());
```

Comment:
i'm not a fan of this, why mixing get and post parameter.

But apart from that, please use array_replace

Reply:
as the test already shows, you send a request with the same parameters and expect the GET to prevail? I would rather expect an exception.

Reply:
Guess the initial issue was that search POST API calls return a response with `links` all containing GET params. Example:

```
{
...
  "links": {
    "first": "https:\/\/shopware.ddev.site\/api\/v3\/search\/product?limit=1&page=1",
    "next": "https:\/\/shopware.ddev.site\/api\/v3\/search\/product?limit=1&page=2",
    "self": "https:\/\/shopware.ddev.site\/api\/v3\/search\/product"
  },
  ...
}
```
Those links would then not work since query parameters were ignored.

Reply:
Asked in the issue about the best solution, since people favoured including query params I provided this PR. Other ideas are welcome :+1:

Reply:
But that is not a real fix. The links are supposed to represent the other pages, but this is not true because they don't take the filters, terms, queries, etc. into account. I rather think we should remove the links when it comes to the /search endpoint

Reply:
Okay, should I update the PR accordingly? I would only execute the logic that adds the pagination info if it is a 'GET' request.